react-accordion-component/my-app/src/Accordion.jsx

```diff
@@ -0,0 +1,43 @@
+import './Accordion.css';
+import { useState } from 'react';
+
+export default function Accordion({ listOfTopics }) {
+  const [current, setCurrent] = useState('');
+
+  function handleCustomClick(clickedButton) {
+    if (clickedButton === current) {
+      setCurrent('');
+    } else {
+      setCurrent(clickedButton);
+    }
+  }
+
+  const topicsArray = [];
+  for (let i = 0; i < listOfTopics.length; i++) {
+    topicsArray.push(
+      <Topic
+        key={listOfTopics[i].id}
+        currentClicked={current}
+        onCustomClick={handleCustomClick}
+        item={listOfTopics[i]}
+      />
+    );
+  }
+
+  return <div>{topicsArray}</div>;
+}
+
+function Topic({ item, onCustomClick, currentClicked }) {
```

Comment:
Instead of passing `currentClicked` and forcing the Topic component to understand the logic the Accordion is using to decide which ones are open, it would be cleaner to pass `isOpen`. Then up in Accordion you do the calculation to decide if it should be open and in this component, you just do what it you're told.